Mathlib/Algebra/Category/AlgebraCat/Basic.lean

```diff
@@ -86,92 +152,54 @@ lemma forget₂_module_obj (X : AlgebraCat.{v} R) :
-
-@[simp]
-theorem id_apply (m : M) : (𝟙 M : M → M) m = m :=
-  rfl
-
-@[simp]
-theorem coe_comp (f : M ⟶ N) (g : N ⟶ U) : (f ≫ g : M → U) = g ∘ f :=
-  rfl
-
-variable (R)
-
 /-- The "free algebra" functor, sending a type `S` to the free algebra on `S`. -/
-@[simps!]
+@[simps! obj map]
 def free : Type u ⥤ AlgebraCat.{u} R where
-  obj S :=
-    { carrier := FreeAlgebra R S
-      isRing := Algebra.semiringToRing R }
-  map f := FreeAlgebra.lift _ <| FreeAlgebra.ι _ ∘ f
-  -- Porting note (https://github.com/leanprover-community/mathlib4/pull/11041): `apply FreeAlgebra.hom_ext` was `ext1`.
-  map_id := by intro X; apply FreeAlgebra.hom_ext; simp only [FreeAlgebra.ι_comp_lift]; rfl
-  map_comp := by
-  -- Porting note (https://github.com/leanprover-community/mathlib4/pull/11041): `apply FreeAlgebra.hom_ext` was `ext1`.
-    intros; apply FreeAlgebra.hom_ext; simp only [FreeAlgebra.ι_comp_lift]; ext1
-    -- Porting node: this ↓ `erw` used to be handled by the `simp` below it
-    erw [CategoryTheory.coe_comp]
-    simp only [CategoryTheory.coe_comp, Function.comp_apply, types_comp_apply]
-    -- Porting node: this ↓ `erw` and `rfl` used to be handled by the `simp` above
-    erw [FreeAlgebra.lift_ι_apply, FreeAlgebra.lift_ι_apply]
-    rfl
+  obj S := of R (FreeAlgebra R S)
+  map f := ofHom <| FreeAlgebra.lift _ <| FreeAlgebra.ι _ ∘ f
+  -- Porting note (#11041): `apply FreeAlgebra.hom_ext` was `ext1`.
+  map_id X := by
+    ext : 1
+    apply FreeAlgebra.hom_ext
+    ext
+    simp
+  map_comp {X Y Z} f g := by
+  -- Porting note (#11041): `apply FreeAlgebra.hom_ext` was `ext1`.
+    ext : 1
+    apply FreeAlgebra.hom_ext
+    ext
+    simp
 
 /-- The free/forget adjunction for `R`-algebras. -/
 def adj : free.{u} R ⊣ forget (AlgebraCat.{u} R) :=
   Adjunction.mkOfHomEquiv
-    { homEquiv := fun _ _ => (FreeAlgebra.lift _).symm
-      -- Relying on `obviously` to fill out these proofs is very slow :(
+    { homEquiv := fun _ _ =>
+        { toFun := fun ⟨f⟩ ↦ (FreeAlgebra.lift _).symm f
+          invFun := fun f ↦ ofHom <| (FreeAlgebra.lift _) f
+          left_inv := fun f ↦ by ext; simp [forget_obj, forget_map]
+          right_inv := fun f ↦ by simp [forget_obj, forget_map]
+        }
       homEquiv_naturality_left_symm := by
-        -- Porting note (https://github.com/leanprover-community/mathlib4/pull/11041): `apply FreeAlgebra.hom_ext` was `ext1`.
-        intros; apply FreeAlgebra.hom_ext; simp only [FreeAlgebra.ι_comp_lift]; ext1
-        simp only [free_map, Equiv.symm_symm, FreeAlgebra.lift_ι_apply, CategoryTheory.coe_comp,
-          Function.comp_apply, types_comp_apply]
-        -- Porting node: this ↓ `erw` and `rfl` used to be handled by the `simp` above
-        erw [FreeAlgebra.lift_ι_apply, CategoryTheory.comp_apply, FreeAlgebra.lift_ι_apply,
-          Function.comp_apply, FreeAlgebra.lift_ι_apply]
-        rfl
+        intros
+        ext : 1
+        apply FreeAlgebra.hom_ext
+        ext
```

Comment:
We could make this:
```suggestion
        ext
```

if we add the following lemma after `free`:
```lean
open FreeAlgebra in
/-- See note [partially-applied ext lemmas]. -/
@[ext high]
theorem free.hom_ext {R X : Type u} {A : Type*} [CommRing R] [Semiring A]
    [Algebra R A] {f g : (free R).obj X →ₐ[R] A}
    (w : (f : FreeAlgebra R X → A) ∘ ι R = (g : FreeAlgebra R X → A) ∘ ι R) : f = g :=
  FreeAlgebra.hom_ext w
```
But I don't think that's worth the effort unless we do that systematically. So this could be left to a follow-up PR.

Reply:
This also works
```
dsimp
ext : 1
ext
simp
```
but we need the manual `ext : 1`.

Reply:
I see! Still not really an improvement, so I think we can leave it as is.